File: app/src/main/java/ru/androidacademy/ateam/model/dao/WordDao.java
package ru.androidacademy.ateam.model.dao;

import android.arch.persistence.room.*;
import android.support.annotation.NonNull;
import io.reactivex.Completable;
import io.reactivex.Flowable;
import io.reactivex.Single;
import ru.androidacademy.ateam.model.tables.Word;

import java.util.List;

@Dao
public interface WordDao {
    @Query("SELECT * FROM Word")
    Single<List<Word>> getAllWords();
    @Query("SELECT COUNT(*) FROM Word")
    int getCountWords();

    @Query("SELECT word FROM word WHERE deckId = :deckId")
    Single<List<String>> getWordByDeckId(@NonNull long deckId);

    @Query("SELECT * FROM Word WHERE deckId = :deckId")
    Single<List<Word>> getAllWordByDeckId(@NonNull long deckId);



    @Query("DELETE FROM word WHERE deckId = :deckId")
    void deleteById(@NonNull long deckId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Word word);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertWordList(List<Word> list);

    @Update
    void update(Word word);

    @Delete
    void delete(Word word);
}

File: app/src/main/java/ru/androidacademy/ateam/db/dao/WordDao.kt
package ru.androidacademy.ateam.db.dao


import androidx.room.*
import io.reactivex.Completable
import io.reactivex.Single
import ru.androidacademy.ateam.model.Word


@Dao
interface WordDao {
    @get:Query("SELECT * FROM Word")
    val getAll: Single<List<Word>>

    @Query("SELECT COUNT(*) FROM word WHERE deckId = :deckId")
    fun getCountByDeckId(deckId: Long): Single<Long>

    @Query("SELECT text FROM word WHERE deckId = :deckId")
    fun getTextByDeckId(deckId: Long): Single<List<String>>

    @Query("SELECT * FROM Word WHERE deckId = :deckId")
    fun getByDeckId(deckId: Long): Single<List<Word>>

    @Query("DELETE FROM word WHERE deckId = :deckId")
    fun deleteById(deckId: Long)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(word: Word)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(list: List<Word>):Completable

    @Update
    fun update(word: Word)

    @Delete
    fun delete(word: Word)

}